### PR TITLE
Entities are stored in SortedDict (for performance)

### PR DESCRIPTION
**Pull request by Makcal**

### CODE REVIEW

**Comment by Makcal** [comment]
Should have been published much earlier

File: src/kikan/engine.py [hunk 1]
```diff
@@ -46,24 +46,27 @@ def _terminate(self):
...
                         entity1, 'collision', [entity2])
                     self.event_manager.dispatch(
                         entity2, 'collision', [entity1])
 
     def _remove_destroyed_entities(self):
-        self.game_world.entities = list(
-            filter(lambda x: not x._is_destroyed, self.game_world.entities))
-        self.game_world.meta_entities = list(
-            filter(lambda x: not x._is_destroyed, self.game_world.meta_entities))
+        for key, value in reversed(self.game_world.entities.items()):
```
Comment (idkarn) [inline src/kikan/engine.py:61]: Why reversed here?

File: src/kikan/events.py [hunk 2]
```diff
@@ -27,19 +27,25 @@ def handle_input(self):
...
     def tick(self):
         self.notify('pre_update')
-        for entity in self.engine.game_world.entities:
+        for entity in self.engine.game_world.entities.values():
             entity._update()
         self.notify('update')
 
         self.handle_input()
 
-    def notify(self, event: EventType, ctx: EventContext = []) -> None:
```
Comment (idkarn) [inline src/kikan/events.py:36]: Why did you remove the `EventContext` type hints?

File: src/kikan/engine.py (hunk 1 above)
Comment (Makcal) [reply src/kikan/engine.py:61]: Modifying a container while iterating it

File: src/kikan/events.py (hunk 2 above)
Comment (Makcal) [reply src/kikan/events.py:36]: Sorry, it's my IDE. I'll fix it
